### DISCUSSION TITLE
ArrayItems 拖动交换顺序与删除前面项的时候如何避免出发值的变化

### DISCUSSION THREAD

**Discussion by biggod-ck** [alibaba/formily#3186] [Q&A 1.x] [answered]
### 问题描述
有一个规则表单。规则名(field),比较符(op),值(value)；这个规则可以一次添加多条。并且当规则名变化后对应的比较符(op)和值(value)需要清空
### 实现代码
[https://codesandbox.io/s/recursing-euclid-4nf15k?file=/App.tsx](https://codesandbox.io/s/recursing-euclid-4nf15k?file=/App.tsx)
### 存在问题
当我拖动改变顺序的时候 与 删除前面的规则会导致表单清空。
### 诉求
希望拖动交换顺序与删除的时候保留表单的值？或者有其他实现方式吗

**Comment by janryWang** [comment, marked as answer]
https://codesandbox.io/s/magical-bird-ots7j7?file=/App.tsx

**Reply by biggod-ck** [reply]
非常感谢大佬的解答！
